kpatch-build/create-diff-object.c

```diff
@@ -70,6 +70,7 @@ enum subsection {
 enum loglevel loglevel = NORMAL;
 
 bool KLP_ARCH;
+bool multi_pfe;
```

Comment:
It's possible that a given toolchain's _intent_ to provide multiple __pfe is not apparent through examining a single ELF, if that ELF has only one patchable function.
So both must be examined before the requirement to emit multi-__pfe style can be determined.
Hence it's not a property of a kelf object, but it could be a property handled within kpatch-elf.c, though with no _use_ made of it there, it seems uglier than the current state.

When @surajjs95 first pushed changes, both his gcc, and google's clang, emitted just a single __pfe, and the code worked for 5.10 in both environments.
Now, latest version of both emit multiples.

From following the discussion of this departure from the old __mcount_loc, thru single-pfe, to the current multi-pfe, in both gcc & clang, I haven't seen a clear signal from the toolchain on which approach is supported, apart from just counting the pfe sections. So that seems to be the best way of accommodating the difference.

Futures?
Whether __pfe or __mcount_loc is used seems to be not so much a per-arch decision as a toolchain decision, given the rationale given by maskray when championing this -- single section demands that whole section be pulled in during link/load when only partial contents needed -- consider pfe entries for .weak functions, where a generic set of fallback C-coded implementations exists to fill the gaps in a per-arch set of optional optimized functions.

With that trajectory in mind, perhaps this might one day become
  enum pfe_style { mcount_loc=0, single_pfe=1,multi_pfe=2};
But later.

Reply:
anyway, I've moved the setting of multi_pfe before its first use, which eliminates the real cause of the problem which manifested as if something was not setting/defining SHF_LINK